x/stake/endblock.go

```diff
@@ -10,9 +10,14 @@ import (
 )
 
 func EndBlocker(ctx sdk.Context, k keeper.Keeper) (validatorUpdates []abci.ValidatorUpdate, completedUbds []types.UnbondingDelegation) {
-	var events sdk.Events
-	_, validatorUpdates, completedUbds, _, events = handleValidatorAndDelegations(ctx, k)
-	ctx.EventManager().EmitEvents(events)
+	// only change validator set in breath block after BEPHHH
+	if !sdk.IsUpgrade(sdk.BEPHHH) {
+		var events sdk.Events
+		_, validatorUpdates, completedUbds, _, events = handleValidatorAndDelegations(ctx, k)
+		ctx.EventManager().EmitEvents(events)
+	} else {
+		k.DistributeInBlock(ctx, keeper.MockSideChainIDForBeaconChain)
```

Comment:
Even before `BEPHHH`, there is no `DistributeInBlock` logic, so  I wonder `DistributeInBlock` is in wrong place right now.

Reply:
I think after `BEPHHH`, we will only do `handleValidatorAndDelegations` in the breath block. There is no related logic for BSC in `EndBlocker` either.